### PR TITLE
build: [INTER-166] use shared workflows

### PR DESCRIPTION
**Pull request by TheUnderScorer**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -17,7 +18,7 @@
     "dist"
   ],
   "engines": {
-    "node": "~16"
+    "node": ">=16"
```
Comment (ilfa) [inline package.json:21]: I'm worried, about why we had the 16th node fixed?

Comment (TheUnderScorer) [reply package.json:21]: Probably to keep local node version the same as lambda runtime, but it causes problems with semantic-release.

Comment (ilfa) [reply package.json:21]: Even when we use semantic release from gh action?

Comment (TheUnderScorer) [reply package.json]: Let's see, I'll revert this change for now

**Pull request merged**